src/main/java/com/iota/iri/service/API.java

```diff
@@ -821,9 +821,11 @@ private synchronized AbstractResponse findTransactionStatement(final Map<String,
     }
 
     private String padTag(String tag) throws ValidationException {
-        while (tag.length() < HASH_SIZE) {
-            tag += Converter.TRYTE_ALPHABET.charAt(0);
+        StringBuilder tagBuilder = new StringBuilder(tag);
+        while (tagBuilder.length() < HASH_SIZE) {
```

Comment:
if this is refactored w/ `StringBuilder` then it makes more sense to just (right) pad the tag to `HASH_SIZE` length, with `9`s. instead of iterating over the length.